### DISCUSSION TITLE
ControlNet Preprocessor Model is missing

### DISCUSSION THREAD

**Discussion by lonnrot** [Mikubill/sd-webui-controlnet#1585] [Q&A]
On the Preprocessor section, after choosing tile_resample, nothing shows up on the Model tab.

Other Preprocessor options have the same result, no model shows up.

This error appears at the same time
```
File "G:\stable-diffusion-webui\venv\Lib\site-packages\gradio\blocks.py", line 1204, in validate_outputs
    raise ValueError(
ValueError: An event handler (run_annotator) didn't receive enough output values (needed: 4, received: 3).
Wanted outputs:
    [image, checkbox, html, html]
Received outputs:
    [{'value': None, 'visible': True, '__type__': 'generic_update'}, {'__type__': 'generic_update'}, {'__type__': 'generic_update'}]

```

Please help, it's been 20 hours trying to get my first conversion to work.

Thanks in advance!
![2023-06-07 16_35_02-Stable Diffusion](https://github.com/Mikubill/sd-webui-controlnet/assets/6014281/8a2bae5d-ac11-488e-a029-ccf1d00ab8e0)

**Comment by lllyasviel** [comment]
Hello, please try the below steps and see if problems are solved:

1. update A1111 and CN to latest version.
2. disable all other A1111 extensions.
3. disable all browser extensions or change another browser.
4. remove all cmd flags in A1111.
5. delete "config.json" and "ui-config.json". (dangerous, remember to backup files)
6. make sure that you have followed the official instruction to download ControlNet models, and make sure that each model is about 1.4 GB large.
7. uninstall ControlNet by removing the controlnet folder and try to install again.
8. make sure to not quit your webui when ControlNet is downloading preprocessor in the background terminal.

The above steps can solve 99% problems. After your problem is solved, you can begin to enable other extensions one-by-one or add A1111 cmd flags one-by-one to find out what is causing problems and let us know.

If you are the unlucky 1%, you can try to use "git checkout" to find out if previous versions of CN works for you, and if that is the case, please let us know your last working version so that we can debug the problem.

If you still cannot solve the problem, please give as detailed as possible info about your environment.

**Reply by lonnrot** [reply]
Got lost in the 4th step.

Which flags and what file? Do you mean a bat file?

**Reply by JefferyWright** [reply]
There do not seem to be any models in the latest to download: [sd-webui-controlnet](https://github.com/Mikubill/sd-webui-controlnet/tree/main/models)

Shouldn't this be a more simple matter of downloading the models manually to sd.webui\webui\models\ControlNet, which is as empty as the one in the repository and the option drop down in the ui? 

So, where are the models?
